soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -49,7 +49,8 @@ extern void z_arm_nmi_init(void);
 #endif
 
 #if (!defined(CONFIG_TRUSTED_EXECUTION_NONSECURE) || defined(CONFIG_BUILD_WITH_TFM)) && \
-	defined(CONFIG_SOC_NRF5340_CPUAPP) && DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_gpio_forwarder)
+	defined(CONFIG_SOC_NRF5340_CPUAPP) && DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_gpio_forwarder) \
+	&& defined(CONFIG_BOARD_ENABLE_CPUNET)
```

Comment:
How about tidying this a bit and using a more logical order? I mean something like:
```
#if DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_gpio_forwarder) &&
	defined(CONFIG_BOARD_ENABLE_CPUNET) &&
	(!defined(CONFIG_TRUSTED_EXECUTION_NONSECURE) ||
	 defined(CONFIG_BUILD_WITH_TFM))
```
I omitted `defined(CONFIG_SOC_NRF5340_CPUAPP)` as it became redundant now.